src/read/macho/exports_trie.rs

```diff
@@ -0,0 +1,269 @@
+use crate::macho::{EXPORT_SYMBOL_FLAGS_REEXPORT, EXPORT_SYMBOL_FLAGS_STUB_AND_RESOLVER};
+use crate::read::{Bytes, ReadError, Result};
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::convert::TryInto;
+
+/// The exports trie is a serialized trie with the following structure:
+///
+/// struct Node {
+///     terminal_size: uleb128, // Size of terminal_data
+///     terminal_data: TerminalData, // Absent if terminal_size == 0
+///     children_count: u8,
+///     edges: [(&[u8] edge_str, uleb128 child_offset); children_count],
+///     children: [Node; children_count],
+/// }
+///
+/// Note that child_offset is relative to the start of the trie data, not the
+/// current node.
+///
+/// TerminalData is a union of the following variants:
+///
+/// struct TerminalDataRegular {
+///     flags: uleb128,
+///     address: uleb128,
+/// }
+///
+/// struct TerminalDataReexport {
+///     flags: uleb128,
+///     dylib_ordinal: uleb128,
+///     import_name: &'data [u8],
+/// }
+///
+/// struct TerminalDataStubAndResolver {
+///     flags: uleb128,
+///     stub_address: uleb128,
+///     resolver_address: uleb128,
+/// }
+///
+/// TerminalData is only present if the current node corresponds to an exported
+/// symbol. Otherwise it is just an internal node in the prefix trie.
+
+/// Iterator over the exports trie.
+#[derive(Debug)]
+pub struct ExportsTrieIterator<'data> {
+    node_iter: NodeIterator<'data>,
+}
+
+impl<'data> ExportsTrieIterator<'data> {
+    pub(super) fn new(data: &'data [u8]) -> Self {
+        ExportsTrieIterator {
+            node_iter: NodeIterator::new(data),
+        }
+    }
+
+    /// Returns the next exported symbol, if any.
+    /// All the heavy lifting is done by NodeIterator. This just skips over the internal nodes
+    /// with no terminal data.
+    fn next(&mut self) -> Result<Option<ExportSymbol<'data>>> {
+        for node in &mut self.node_iter {
+            if let Some(terminal_data) = node? {
+                return Ok(Some(ExportSymbol {
+                    name: self.node_iter.name_buf.clone().into_boxed_slice(),
```

Comment:
I feel like we should return a reference instead, but I guess that prevents an `Iterator` impl.